### ISSUE TITLE
How to test the performance of the python tutorial example?

### ISSUE THREAD

**Issue by ElizaTay** [mGalarnyk/Python_Tutorials#4]

**Comment by mGalarnyk** [comment]
What do you mean by that? Test to see if something is working (unit tests?). 

Which tutorial in particular are you interested in?

**Comment by mGalarnyk** [comment]
Also if you find an issue with something let me know and feel free to submit a pull request :) I haven't had any yet for this repo despite it being around for a long time.

**Comment by ElizaTay** [comment]
you submit very good code called "Python_Tutorials". I rely understand some points from your example. Now if I want to test the performance should I guess some data and see the result is the target or not?.
after that I have to measure accuracy,precision and recall.

**Comment by ElizaTay** [comment]
I have some questions here if you please help me I appreciate that 
first question 
y=diabetes.target
does this portion of the code means i am getting the target or classes of the features? when I ran it it did not give me the target
***
second question 
predicted=cross_val_predict(model,x,y,cv=10)
does this portion of code apply the model on the train set , then predict the classes of the test set ? if so how I can visualize both sets?

**Comment by mGalarnyk** [comment]
show me the code you wish to emulate.

**Comment by mGalarnyk** [comment]
the entire code.

**Comment by ElizaTay** [comment]
It is simple code , but I am confused because i am beginner 
The comments is either the step that I wanna do ,or some previous trails

```
from sklearn import datasets
from sklearn.model_selection import cross_val_predict
from sklearn.model_selection import cross_val_score
from sklearn import linear_model
import matplotlib.pyplot as plt
from sklearn.model_selection import KFold
model=linear_model.LinearRegression()
diabetes=datasets.load_diabetes()
##step1 : read the diabets dataset
x = diabetes.data
y = diabetes.target

### step2: Divide the data set to train and test
### stept3: train the linear regression
predicted=cross_val_predict(model,x,y,cv=10)
### stept4: Test the performance
#testdata=[0.038,0.05,0.06,0.023,0.089,0.032,0.067,0.014,0.022,0.088]
#test=cross_val_predict(model,testdata,cv=10)
### stept5: report accuracy,recall (not yet)
#print 'Accuracy:', accuracy_score(y_test, prediction) (not yet)
### stept6: report most significant features in the regression model (not yet)
### stept7: plot the relation between feature xi and target y
fig, ax = plt.subplots()
ax.scatter(y,predicted)
ax.plot([y.min(),y.max()],[y.min(),y.max()],'k--',lw=4)
ax.set_xlabel('Actual y')
ax.set_xlabel('Predicted y')
results = cross_val_score(model, x, y, cv=10)
#from sklearn.metrics import classification_report
#print(classification_report(y,predicted))
print(results)
plt.show()
#kf = KFold(n_splits=10)
     #kf.get_n_splits(x)
     #print(kf)
    #for train_index, test_index in kf.split(x):
     #print("TRAIN:", train_index, "TEST:", test_index)
    # x_train, x_test = x[train_index], x[test_index]
   #y_train, y_test = y[train_index], y[test_index]
```

**Comment by mGalarnyk** [comment]
Not related to a tutorial here. I will maybe answer it in a future blog post.
